src/ChatBubble/ChatBubble.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+  ReactNode,
+  CSSProperties,
+  useCallback,
+  useMemo,
+  memo,
+  useState,
+  useRef,
+  useEffect
+} from 'react';
+import { createPortal } from 'react-dom';
+import { cn } from 'reablocks';
+import { motion, AnimatePresence } from 'motion/react';
+
+export type Position =
+  | 'bottom-left'
+  | 'bottom-right'
+  | 'top-left'
+  | 'top-right';
+
+export interface ChatBubbleProps {
+  /**
+   * The main content to be rendered.
+   */
+  children: ReactNode;
+
+  /**
+   * The content to be rendered in the trigger bubble.
+   */
+  bubbleContent: ReactNode;
+
+  /**
+   * The position of the chat bubble on the screen.
+   * @default 'bottom-left'
+   */
+  position?: Position;
+
+  /**
+   * Custom CSS styles to override the default positioning.
+   */
+  customPosition?: CSSProperties;
+
+  /**
+   * The DOM element where the chat bubble should be rendered.
+   * @default document.body
+   */
+  portalTarget?: HTMLElement | null;
+
+  /**
+   * Additional CSS classes to apply to the chat bubble.
+   */
+  className?: string;
+}
+
+const defaultPositions: Record<Position, string> = {
+  'bottom-left': 'bottom-5 left-5',
+  'bottom-right': 'bottom-5 right-5',
+  'top-left': 'top-5 left-5',
+  'top-right': 'top-5 right-5'
+};
+
+export const ChatBubble = memo<ChatBubbleProps>(
+  ({
+    children,
+    bubbleContent,
+    position = 'bottom-left',
+    customPosition,
+    portalTarget = typeof document !== 'undefined' ? document.body : null,
+    className
+  }) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const bubbleRef = useRef<HTMLDivElement>(null);
+    const contentRef = useRef<HTMLDivElement>(null);
+    const [bubbleRect, setBubbleRect] = useState<DOMRect | null>(null);
+
+    useEffect(() => {
+      if (bubbleRef.current) {
+        setBubbleRect(bubbleRef.current.getBoundingClientRect());
+      }
+    }, [isOpen]);
+
+    const handleClose = useCallback(() => {
+      setIsOpen(false);
+    }, []);
+
+    const handleToggle = useCallback(() => {
+      setIsOpen(prev => !prev);
+    }, []);
+
+    const getContentPosition = () => {
+      if (!bubbleRect) return {};
+
+      const positions: Record<Position, CSSProperties> = {
```

Comment:
I believe this could simplify some of this logic and make it more robuts - https://reablocks.dev/docs/utils/overlay